udata/core/reports/models.py

```diff
@@ -42,15 +47,31 @@ class Report(db.Document):
 
     @classmethod
     def mark_as_deleted_soft_delete(cls, sender, document, **kwargs):
-        if document.deleted:
-            Report.objects(
-                object_type=sender.__name__, object_id=document.id, object_deleted_at=None
-            ).update(object_deleted_at=datetime.utcnow)
+        """
+        Called when updating a model (maybe updating the `deleted` date)
+        Some documents like Discussion do not have a `deleted` attribute.
+        """
+        if hasattr(document, "deleted") and document.deleted:
+            # It's a little bit hard to query the GenericReferenceField,
+            # we could do it without extra request with `DBRef(sender.__name__.lower(), document.id)`
+            # but I'm not a big fan of the `.lower()` to get the correct DBRef. Fetching the model
+            # and asking MongoDB for the DBRef with `.to_dbref()` seems more robust.
+            subject = sender.objects(id=document.id).first()
+            Report.objects(subject=subject.to_dbref(), subject_deleted_at=None).update(
+                subject_deleted_at=datetime.utcnow
+            )
```

Comment:
I'm confused here. Why doesn't it work with a straight:
```
Report.objects(subject=document)
```
This is how I would go for discussion for example.

Reply:
Done in ea5d8ef6e. Don't know why it wasn't working before… :-(